src/lib.rs

```diff
@@ -588,11 +632,13 @@ pub fn run_for_slice<'a>(
         },
     );
     drop(from_file_run_context_instance_provider);
-    (violations.into_inner().unwrap(), instantiated_rules)
-    // RunForSliceStatus {
-    //     violations: violations.into_inner().unwrap(),
-    //     from_file_run_context_instance_provider,
-    // }
+    SliceRunStatus {
+        violations: violations.into_inner().unwrap(),
+        per_config_context: match per_config_context {
+            Cow::Borrowed(_) => None,
+            Cow::Owned(per_config_context) => Some(per_config_context),
```

Comment:
The idea here is that in can be treated (by the LSP server) as the "contract" of this function that it will definitely populate/pass back `PerConfigContext` if it didn't get passed in a reference to an existing `PerConfigContext` (enabling the LSP server to then "confidently" unwrap/cache that newly initialized `PerConfigContext` at the end of the first linting run it does)